### PR TITLE
Quizzes by category and rules

### PR DESCRIPTION
**Pull request by partha8**
## Note
CSS can be ignored.

## Preview
image

image

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -19,11 +20,19 @@ const App = () => {
...
+        </header>
         <Routes>
           <Route path="/" element={<Home />} />
+          <Route path="/category/:categoryName" element={<QuizCategory />} />
+          <Route element={<RuleRouting />}>
+            <Route path="/quiz/:quizId" element={<QuestionPage />} />
+          </Route>
         </Routes>
+        <Footer />
       </Router>
```
Comment (Dev-AkshatGupta) [inline src/App.tsx:34]: Suggestion take this whole Router in a seperate component

File: src/components/Footer/Footer.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import { FaGithub, FaTwitter, FaLinkedin } from "react-icons/fa";
+import { AiOutlineGlobal } from "react-icons/ai";
+import "./footer.css";
+
+export const Footer = () => {
+  return (
+    <footer>
+      <a href="https://twitter.com/partha_sarma8" target="__blank">
+        <FaTwitter />
```
Comment (Dev-AkshatGupta) [inline src/components/Footer/Footer.js]: Suggestion keep the whole application in either js or ts

File: src/components/RuleRouting/RulesRouting.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { RuleModal } from "../RuleModal/RuleModal";
+import { useState } from "react";
+import { Outlet } from "react-router-dom";
+
+export const RuleRouting = () => {
+  const [showRules, setShowRules] = useState<boolean>(true);
+  return showRules ? (
+    <RuleModal showRules={showRules} setShowRules={setShowRules} />
+  ) : (
+    <Outlet />
```
Comment (Dev-AkshatGupta) [inline src/components/RuleRouting/RulesRouting.tsx:10]: Nice that you are using the Outlet

**Review by Dev-AkshatGupta — APPROVED** [review]
Everything looks great to me just gave couple of suggestions

File: src/components/Footer/Footer.js (hunk 2 above)
Comment (partha8) [reply src/components/Footer/Footer.js]: arey haaa! I missed this since i copy pasted it from other project. Thank you!

Comment (partha8) [reply src/components/Footer/Footer.js:10]: arey haaa! I missed this since i copy pasted it from other project. Thank you!

**Pull request merged**